Src/WitsmlExplorer.Frontend/tools/localStorageHelpers.tsx

```diff
@@ -0,0 +1,140 @@
+      }
+      return parsedItem || defaultValue || null;
+    }
+  } catch (error) {
+    if (error instanceof SyntaxError) {
+      console.warn(`Error parsing localStorage item for key “${fullKey}”. Removing the item from localStorage as the type might have changed.`, error);
+      removeLocalStorageItem(key, options);
+    } else {
+      // disregard unavailable local storage
+      console.warn(`Error getting localStorage item for key “${fullKey}”:`, error);
+    }
+  }
+  return defaultValue || null;
+};
+
+export const setLocalStorageItem = <Key extends StorageKey>(key: Key, value: StorageKeyToType[Key], options?: storageOptions<Key>): void => {
+  const { preKey, storageTransformer } = options || {};
+  const fullKey = preKey ? preKey + key : key;
+  try {
+    if (typeof window !== "undefined" && fullKey) {
+      const transformedValue = storageTransformer ? storageTransformer(value) : value;
+      localStorage.setItem(fullKey, JSON.stringify(transformedValue));
+    }
+  } catch (error) {
+    // disregard unavailable local storage
+    console.warn(`Error setting localStorage item for key “${fullKey}”:`, error);
+  }
+};
+
+export const removeLocalStorageItem = <Key extends StorageKey>(key: Key, options?: storageOptions<Key>): void => {
+  const { preKey } = options || {};
+  const fullKey = preKey ? preKey + key : key;
+  try {
+    if (typeof window !== "undefined" && fullKey) {
+      localStorage.removeItem(fullKey);
+    }
+  } catch (error) {
+    // disregard unavailable local storage
+    console.warn(`Error removing localStorage key “${fullKey}”:`, error);
+  }
+};
+
+/**
+ * Interface for the options object that can be passed to the localStorage helper functions.
+ *
+ * @interface
+ * @template Key - A key from the StorageKey enum.
+ *
+ * @property preKey - An optional prefix for the localStorage key.
+ * @property defaultValue - An optional default value to use when the localStorage item is not found or an error occurs.
+ * @property delay - An optional delay (in milliseconds) used to debounce the setting of the item. Only used by the useLocalStorageState hook.
+ * @property valueVerifier - An optional function to verify the value retrieved from localStorage. If this function returns false, defaultValue or null will be used instead.
+ * @property storageTransformer - An optional function to transform the value before it's stored in localStorage.
+ */
+export interface storageOptions<Key extends StorageKey> {
+  preKey?: string;
+  defaultValue?: StorageKeyToType[Key];
+  delay?: number;
+  valueVerifier?: (value: StorageKeyToType[Key]) => boolean;
+  storageTransformer?: (value: StorageKeyToType[Key]) => StorageKeyToType[Key];
+}
+
+export const useLocalStorageState = <Key extends StorageKey>(
+  key: Key,
+  options?: storageOptions<Key>
+): [StorageKeyToType[Key] | null, React.Dispatch<React.SetStateAction<StorageKeyToType[Key] | null>>] => {
+  const { preKey, defaultValue, delay = 250 } = options || {};
```

Comment:
Can the preKey logic be separated from the `useLocalStorageState` hook?

Reply:
Done